scripts/nn/test/test.dml

```diff
@@ -605,3 +605,20 @@ tanh = function() {
   }
 }
 
+threshold = function() {
+    /*
+     * Test for threshold function.
+    */
+    print("Testing the threshold function.")
+
+    # Generate data
+    matrix = matrix("0.31 0.24 0.87
+
+                    0.45 0.66, 0.65
+
+                    0.24 0.91 0.13", rows=3, cols=3)
+    thresh = 0.5
+
+    # Get the indicator matrix
+    indicator_matrix = util::threshold(matrix, thresh)
```

Comment:
This is good so far, but we need to add a `target` matrix with the expected output for this example, and then compare `target` with `indicator_matrix` using the `util::check_all_equal` function (located in `nn/test/util.dml`) to test that this function is working correctly.  It can be helpful to purposely make it wrong temporarily to ensure that the testing code itself is working.